File: packages/emitter/testcases/common/types/expected-type-threading/ArraySpread.ts
import { int } from "@tsonic/core/types.js";

// Array spread with int type - spread elements get array type
const source: int[] = [1, 2, 3];
export const withSpread: int[] = [...source, 4, 5];

// Multiple spreads
const more: int[] = [10, 20];
export const multiSpread: int[] = [...source, ...more, 100];

File: packages/emitter/testcases/common/expected/types/expected-type-threading/ArraySpread.cs
namespace TestCases.common.types.expectedtypethreading
{
    [global::Tsonic.Internal.ModuleContainerAttribute]
    public static class ArraySpread
    {
        internal static readonly int[] source = new int[] { 1, 2, 3 };

        public static readonly int[] withSpread = global::System.Linq.Enumerable.ToArray(global::System.Linq.Enumerable.Concat(source, new int[] { 4, 5 }));

        internal static readonly int[] more = new int[] { 10, 20 };

        public static readonly int[] multiSpread = global::System.Linq.Enumerable.ToArray(global::System.Linq.Enumerable.Concat(global::System.Linq.Enumerable.Concat(source, more), new int[] { 100 }));
    }
}
